### PR TITLE
fix(metrics): pass by class reference rather than instantiated class DEVX-585

### PR DESCRIPTION
**Pull request by nsheaps**
## 🧪 Testing Steps / Validation
I pulled the local package into a local pip install into moderators, which exhibited the problem, then added some logging to confirm that the qsize != size at which point it was obvious that they were all sharing one queue. 

## ✅ Checks

- [x] CI tests (if present) are passing
- [x] Adheres to code style for repo
- [ ] Contributor License Agreement (CLA) completed if not a Reddit employee

### CODE REVIEW

**Comment by RoganMurley** [comment]
Good catch, looks like this was broken in #645

**Review by RoganMurley — APPROVED** [review]

File: baseplate/lib/thrift_pool.py [hunk 1]
```diff
@@ -170,7 +171,7 @@ def __init__(
 
         self.size = size
 
-        self.pool = queue_cls
+        self.pool = queue_cls()
```
Comment (mterwill) [inline baseplate/lib/thrift_pool.py:174]: I can't find anywhere internally that `queue_cls` is being passed in as an argument, but just wanted to call out that this is technically a backwards-incompatible change.

**Review by mterwill — APPROVED** [review]

**Comment by a--hoang** [comment]
Note that in addition to metrics, this will also possibly cause thrift connection issues in the case that you've instantiated more than one thrift client in your application and used the default params. 

This will result in all thrift clients in your application using the SAME queue, which means that they can accidentally grab each others' connections/sockets and send invalid requests to the wrong server. All clients are also in contention with each other for the pool size, so increased timeouts/connection failures are likely.

Glad we got this fixed - hoping we can do a bump to v2.5.x after this is merged since this potentially has large impact

**Review by KTAtkinson — APPROVED** [review]
Good find!

Please make sure to not reference Reddit internal services in PR descriptions or commit messages.

File: baseplate/lib/thrift_pool.py (hunk 1 above)
Comment (nsheaps) [reply baseplate/lib/thrift_pool.py:174]: You are correct. Only 3 services are on 2.4.x+ and none of them take advantage of this (this is primarily used for tests)

**Pull request merged**